concourse/scripts/pipelines-bosh-cloudfoundry.sh

```diff
@@ -116,38 +116,47 @@ generate_manifest_file() {
     < "${SCRIPT_DIR}/../pipelines/${pipeline_name}.yml"
 }
 
+upload_pipeline() {
+  bash "${SCRIPT_DIR}/deploy-pipeline.sh" \
+        "${env}" "${pipeline_name}" \
+        <(generate_manifest_file) \
+        <(generate_vars_file)
+}
+
+remove_pipeline() {
+  yes y | ${FLY_CMD} -t "${FLY_TARGET}" destroy-pipeline --pipeline "${pipeline_name}" || true
+}
+
 update_pipeline() {
   pipeline_name=$1
 
   case $pipeline_name in
-    create-bosh-cloudfoundry|failure-testing)
-      bash "${SCRIPT_DIR}/deploy-pipeline.sh" \
-        "${env}" "${pipeline_name}" \
-        <(generate_manifest_file) \
-        <(generate_vars_file)
+    create-bosh-cloudfoundry)
+      upload_pipeline
+    ;;
+    failure-testing)
+      if [ -n "${ENABLE_FAILURE_TESTING:-}" ] && [ "${ENABLE_FAILURE_TESTING:-}" = "true" ]; then
+        upload_pipeline
+      else
+        remove_pipeline
+      fi
     ;;
     destroy-*)
-      if [ -n "${ENABLE_DESTROY:-}" ]; then
-        bash "${SCRIPT_DIR}/deploy-pipeline.sh" \
-          "${env}" "${pipeline_name}" \
-          <(generate_manifest_file) \
-          <(generate_vars_file)
+      if [ -n "${ENABLE_DESTROY:-}" ] && [ "${ENABLE_DESTROY:-}" = "true" ]; then
```

Comment:
This is a change in behaviour (that's not called out in the PR summary). It also makes these variables inconsistent with some other variables in the `Makefile` (eg `SKIP_COMMIT_VERIFICATION`).  If we're going to change how these boolean flags are interpreted, I'd be keen that we do it for all of them.